### PR TITLE
Always show scrollbar regardless of content height

### LINKED ISSUE #30705: Page Layout/Margins not consistent across all pages when viewed on wide enough screen
### Preconditions (*)

1. 2.4.1 and 2.4-develop
2. The same behavior on Blank and Luma Themes

### Steps to reproduce (*)

1. Deploy a blank vanilla site
2. Expand browser window so there are left and right margins on screen with site centered
3. Create a test product in a test category
4. Browse the following pages, noticing if the position of the logo/navbar/content changes on each page from left/right by a small amount

- Home Page
- Category Page
- Product Page
- Popular Search Terms
- Orders and Returns
- Contact Us
- Advanced Search

### Expected result (*)

1. Page Layouts should be consistent

### Actual result (*)

1. They are not
2. Sometimes the content shifts right and then back again very quickly, other times it stays offset.

8fZgdqCovk

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

This PR is improvements for blank/luma theme 

#1 Add some Improvements for blank theme
#2 Default specific width height default for logo via layout but owen can easy override dimensions from backend configuration
#3 Improve magento _reset styles default

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30705

### Manual testing scenarios (*)

Above screen 768px
1. View category page, contact page, advanced search
2. Logo will not shift from right to left like before when page reload
We should test for both theme luma and blank
but luma has high priority than blank because owner report issue in luma

### Questions or comments

CC: @krzksz @ptylek @Leland feel free send me any feedback
@gwharton 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
Hi @mrtuvn,
How this change could be tested?

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

File: app/design/frontend/Magento/blank/web/css/source/_layout.less [hunk 1]
```diff
@@ -93,7 +93,14 @@
...
 
     .page-main {
+        .lib-vendor-prefix-flex-grow(1);
+        .lib-vendor-prefix-flex-shrink(0);
+        .lib-vendor-prefix-flex-basis(auto);
         width: 100%;
+
+        .ie9 & {
+            width: auto;
+        }
```
Comment (ihor-sviziev) [inline app/design/frontend/Magento/blank/web/css/source/_layout.less]: I believe we don't need to support ie9, so this should be removed.

Comment (hostep) [reply app/design/frontend/Magento/blank/web/css/source/_layout.less]: Yeah indeed, Magento dropped support for Internet Explorer completely with the release of Magento 2.4.0

Docs: [2.3.x browser support](https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements.html#supported-browsers) vs [2.4.x browser support](https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements.html#supported-browsers)

**Comment by ihor-sviziev** [comment]
Cool! Thank you for the fix!

@magento run all tests

File: app/design/frontend/Magento/blank/web/css/source/_layout.less (hunk 1 above)
Comment (mrtuvn) [reply app/design/frontend/Magento/blank/web/css/source/_layout.less]: I have removed that

**Comment by ihor-sviziev** [comment]
@sidolov could you add some info why Integration tests failing?

**Comment by sidolov** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@ihor-sviziev I didn't find the cause, let's run them once again

**Comment by ihor-sviziev** [comment]
@magentor run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests looks not related to changes form this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8452](https://jira.corp.magento.com/browse/ENGCOM-8452) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
@magento give me test instance

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @mrtuvn .

It looks like the changes don't solve the problem.

**Actual Result:** :heavy_multiplication_x:  shift from right to left like before when page reload

Peek 2020-11-12 15-32
Peek 2020-11-12 15-35

@mrtuvn Could you take a look?
Thank you!

**Comment by ihor-sviziev** [comment]
As you can see - it’s caused by the scroll bar that appears on the right
side

**Comment by mrtuvn** [comment]
yes scrollbar appear will make layout move slightly a little bit

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

File: app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml [hunk 2]
```diff
@@ -8,10 +8,5 @@
 <page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
     <head>
         <meta name="viewport" content="width=device-width, initial-scale=1"/>
-        <font src="fonts/opensans/light/opensans-300.woff2"/>
-        <font src="fonts/opensans/regular/opensans-400.woff2"/>
-        <font src="fonts/opensans/semibold/opensans-600.woff2"/>
-        <font src="fonts/opensans/bold/opensans-700.woff2"/>
-        <font src="fonts/Luma-Icons.woff2"/>
```
Comment (ihor-sviziev) [inline app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: those included fonts were added here in order to have some font preload that google pagespeed is telling for. Are you sure it's causing  layout shift?

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

File: app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml (hunk 2 above)
Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: @ihor-sviziev i'm not sure why this located here. But if you remove it icon still show exactly like before. We can safe remove this

Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: Previous this from blank theme. After some commit it's moved to luma. Default icon and font imported via less

Comment (ihor-sviziev) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: Yes, they are imported from less files, but it adds font preload, like this:
```html
<link rel="preload" as="font" crossorigin="anonymous" href="https://www.yourdomain.com/static/version1605168925/frontend/Magento/Luma/en_US/fonts/Luma-Icons.woff2" />
```

Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: I think this is unnecessary. It Take more request to browser

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

File: app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml (hunk 2 above)
Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content
Should i need revert this change back to previous ?
Preloading these fonts ensures they are available earlier and are less likely to block the page's render, improving performance

Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml:15]: cc @krzksz what do you think

Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Theme/layout/default_head_blocks.xml]: Reverted this back to previously

**Comment by mrtuvn** [comment]
rebased + combine commits to one

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
Fail tests seem not related with changes

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8452](https://jira.corp.magento.com/browse/ENGCOM-8452) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Deploy a luma or blank vanilla site
2. Expand browser window so there are left and right margins on screen with site centered
3. Create a test product in a test category
4. Browse the following pages, noticing if the position of the logo/navbar/content changes on each page from left/right by a small amount

- Home Page
- Category Page
- Product Page
- Popular Search Terms
- Orders and Returns
- Contact Us
- Advanced Search

**Before:** :heavy_multiplication_x: Sometimes the content shifts right and then back again very quickly, other times it stays offset.

Peek 2020-11-17 13-09

**After:** :heavy_check_mark: Page Layouts are consistent

**_Luma_**

Peek 2020-11-17 13-00

**_Blank_**

Peek 2020-11-17 13-04

**Comment by ihor-sviziev** [comment]
@engcom-Alfa what about mobile?

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by mrtuvn** [comment]
@ihor-sviziev just do some check in mobile with lighthouse score . Not quite good score in below mobile

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Functional Tests CE, Integration Tests, WebAPI Tests

**Comment by mrtuvn** [comment]
Ok it's make sense now after previous day i got so many break in tests report from magento

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run Integration Tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B, Integration Tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests EE, Functional Tests B2B, Integration Tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
I checked - changes still the same, they were just rebased on top of recent 2.4-develop branch, so still approved from my side.

@magento run Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento give me 2.4-develop instance

**Comment by mrtuvn** [comment]
@magento give me 2.4-develop instance

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Changes still the same
@magento run all tests

**Comment by mrtuvn** [comment]
rebase code base

**Comment by mrtuvn** [comment]
rebase branch PR

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento give me 2.4-develop instance

**Comment by ihor-sviziev** [comment]
Hi @mrtuvn,
Could you ping me when you'll finish all the wanted updates and I can start reviewing the pull request?

**Comment by mrtuvn** [comment]
yes can we have proceed review on this

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
looks like all the test failures are false positive🤦‍♂️\

@magento run Functional Tests B2B, Functional Tests EE, Integration Tests, Unit Tests, WebAPI Tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by ihor-sviziev** [comment]
@mrtuvn are you going to change something in your solution?

**Comment by mrtuvn** [comment]
@ihor-sviziev there are some problem i need to re-check. Since previous change make some buggy problem while i tested in my local. So i keep this is draft for test

**Comment by mrtuvn** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]
✅ Approved
Failing functional tests looks not related to changes from this PR

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8452](https://jira.corp.magento.com/browse/ENGCOM-8452) has been created to process this Pull Request

**Comment by Leland** [comment]
> **Fix layout shift while render**

Corrected PR title, IMO:

> **Always show scrollbar regardless of content height**

Can't that say I agree with this PR, or the linked issue, #30705.

This is addressing a non-existent problem, which is, succinctly, that scrollbars show on tall pages but not short ones.

That's just how the web works. If a page is short, it shouldn't show scrollbars. This also is not a case of Cumulative Layout Shift, as the "shift" only happens when navigating between pages - not during the lifecycle of a single page.

**Comment by mrtuvn** [comment]
Thank you for some advices and feedbacks

**Comment by mrtuvn** [comment]
Do we need this additional ? Added some improvements in magento default _reset.less body tag avoid shifty experience

**Comment by Leland** [comment]
My personal position is that that shift is completely normal and does not need addressing.

Forcing scrollbars to always show is, IMO, something most website owners would not want.

File: lib/web/css/source/lib/_resets.less [hunk 3]
```diff
@@ -12,9 +12,20 @@
...
         margin: 0;
+        overflow-x: auto;
+        overflow-y: scroll;
+        overscroll-behavior-y: none;
         padding: 0;
+        scroll-behavior: smooth;
+        width: 100%;
     }
 
     article,
```
Comment (mrtuvn) [inline lib/web/css/source/lib/_resets.less:26]: @Leland do we need this additional ?

**Comment by mrtuvn** [comment]
rebased code and minimal number files changes to avoid conflicts and unintended issues

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by mrtuvn** [comment]
i'm close this solution because this is actually normal problem. That's how browser work

**Pull request closed without merging**